data/api/client-server/definitions/sync_filter.yaml

```diff
@@ -16,10 +16,9 @@ title: Filter
 properties:
   event_fields:
     description: List of event fields to include. If this list is absent then all
-      fields are included. The entries may include '.' characters to indicate sub-fields.
-      So ['content.body'] will include the 'body' field of the 'content' object. A
-      literal '.' character in a field name may be escaped using a '\\'. A server may
-      include more fields than were requested.
```

Comment:
The MSC doesn't *quite* say this since this implicitly adds escaping a backslash with another backslash, which I don't think was previously valid. I can back this out if you would like.

Reply:
Hmm, since MSC3873 doesn't propose modifying sync filters to support escaping backslashes (instead just taking the dot-escaping capability of sync filter), I believe the *correct* approach to unify the two would be to have another MSC about Sync filters to include support for escaping backslashes.

I think such an MSC would be met without any contention, as everyone just wants consistency at the end of the day.

However, while that separate MSC is in review and MSC3873 is merged, it does mean that clients/servers will have to match backslashes differently depending on whether they're processing push rules or sync filters, which is a bit of a pain. So this may be a case of the process getting in the way, and we simply edit MSC3873 after the fact to mention that sync filters should also escape backslashes.

Finally, this is all a bit moot as no fields in events currently contain backslashes anyway (except possibly some displaynames?).

What do other @matrix-org/spec-core-team members think?

Reply:
> Finally, this is all a bit moot as no fields in events currently contain backslashes anyway (except possibly some displaynames?).

It only applies to property *names*, not values.

Reply:
I decided to just write the MSC: see [MSC3980](https://github.com/matrix-org/matrix-spec-proposals/pull/3980).

Reply:
I'll try to push for review of the MSC on Tuesday with the rest of the Spec Core Team as it should be universally agreeable and a very short read. Thank you for writing it :slightly_smiling_face:

Reply:
@anoadragon453 Re-requested review, although this is blocked on the FCP timer of MSC3980.

Reply:
This should be ready to go now. 👍